clash-prelude/src/Clash/Sized/RTree.hs

```diff
@@ -537,3 +544,63 @@ lazyT :: KnownNat d
       => RTree d a
       -> RTree d a
 lazyT = tzipWith (flip const) (trepeat ())
+
+-- | Low-depth (left) scan, see 'Clash.Sized.Vector.scanl'.
+--
+-- >>> scanlPar (+) (1 :> 2 :> 3 :> 4 :> Nil)
+-- 1 :> 3 :> 6 :> 10 :> Nil
+--
+-- The operator must be associative.
+--
+-- <<doc/scanlPar.svg>>
+scanlPar :: KnownNat n
+         => (a -> a -> a) -- ^ Must be associative
+         -> Vec (2^n) a -> Vec (2^n) a
```

Comment:
```suggestion
scanlPar :: 
  KnownNat n =>
  -- | Must be associative
  (a -> a -> a) ->
  Vec (2^n) a ->
  Vec (2^n) a
```

We've switched to [Ormolu's style](https://yairchu.github.io/posts/silly-haskell-formatting) oone or two releases back.

Reply:
It fits one line btw, I wouldn't mind losing the argument doc as it's explained in the function doc anyway.